internal/io/http/client_edgex.go

```diff
@@ -0,0 +1,63 @@
+//go:build edgex
+
+/*
+Copyright NetFoundry Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"net/http"
+
+	"github.com/openziti/sdk-golang/ziti"
+	"github.com/sirupsen/logrus"
+
+	"github.com/lf-edge/ekuiper/internal/conf"
+	edgex_vault "github.com/lf-edge/ekuiper/internal/edgex"
+)
+
+func init() {
+	newTransport = newZeroTrustTransport
+}
+
+var zitiTransport *http.Transport
+
+func newZeroTrustTransport(tlscfg *tls.Config, logger *logrus.Logger) *http.Transport {
```

Comment:
When using EdgeX secure mode, is the zero trust transport applied to all http request or just apply to the request for EdgeX url? If the latter, we don't need to always return this transport. It should be a property in the REST sink and only those sink with this property need to get the zero trust transport.

Reply:
It depends on how the transport is returned. the zero trust transport needs to only intercept the addresses it's configured to intercept but it should definitely fallback to using the regular underlay if the url is not one it's expected to intercept. So it should work in a "split-tunnel" type of way. 

Thanks for asking -- I tested to verify that would be the case and it's not currently. I'll fix that today.

Reply:
I've changed the transport to use a fallback dialer which will activate if no matching service is found